### DISCUSSION TITLE
Embedding tectonic for in-application TeX rendering

### DISCUSSION THREAD

**Discussion by fpdotmonkey** [tectonic-typesetting/tectonic#1133] [General]
I'm curious if tectonic can be used to render TeX input not to a PDF, but for example to a texture that could then be rendered in a GUI application, or perhaps as an SVG to the same effect.

I'm interested in this to produce something like the fancy mathy motion graphics you see on YouTube these days, and showing nicely rendered equations are a part of that.

**Comment by pkgw** [comment]
To the best of my knowledge, the current answer is ... yes, but with difficulty and/or inefficiency.

You can have Tectonic produce a PDF output and the process it with something like [`pdf2svg`][1] or [`pdftocairo`][2] to get an SVG, PNG, or other output. This isn't super elegant/efficient, and might not be an option at all in some scenarios since it involves executing separate programs. But it definitely works, and is what I do when I need to generate slides with equations, since I deliver my talks in HTML. I usually do this with shell scripts, but with a bit of work to use tempfiles and such, you can almost surely deliver this kind of conversion as a callable function in your language of choice.

[1]: https://github.com/dawbarton/pdf2svg
[2]: https://manpages.ubuntu.com/manpages/trusty/man1/pdftocairo.1.html

More efficient, but probably *much* more work to implement, you could parse Tectonic's PDF output in-memory, or the the intermediate "XDV" format that is emitted by the core XeTeX engine. The [tectonic_xdv][3] crate can parse the latter format,
and the [tectonic_engine_spx2html][4] crate is an example that does that to generate an alternative output (specialized HTML rather than PDF; note that the work in that crate is *very* preliminary and not a generic TeX-to-HTML converter).

[3]: https://github.com/tectonic-typesetting/tectonic/tree/master/crates/xdv
[4]: https://github.com/tectonic-typesetting/tectonic/tree/master/crates/engine_spx2html

**Reply by fpdotmonkey** [reply]
Interesting, I'll look into these
